docs/content/schemas.md

```diff
@@ -108,6 +113,46 @@ Consider the following input document:
 
   This indicates the error to the Rego developer right away, without having the need to observe the results of runs on actual data, thereby improving productivity.
 
+## Using annotations on Rules to further enhance the Rego type checker
+
+A rule can be annotated with a comment of the form:
+
+#@rulesSchema=`<expression>:<path-to-schema>,...,<expression>:<path-to-schema>`
+
+An expression is of the form `<input|data>.field1. ... .fieldN`
+
+This annotation associates a schema (provided via OPA's `opa eval --schema`) with the corresponding expression. So it can be used to give a schema to the input or any data document. The type checker derives a Rego Object type for the schema and an appropriate entry is added to the type environment. This entry is removed upon exit from the rule.
+
+Annotations allow overriding when a prefix of an expression has a type in the type environment (see example below).
+
+Notice that currently the annotation needs to appear on the line immediately preceding the rule definition.
+
+Examples:
+
+Consider a directory named `schemas` with the following structure, uploaded via `opa eval --schema /Users/ansuvarghese/opa-schema-examples/kubernetes/schemas`
```

Comment:
Same as above, this "uploaded" is a bit confusing to me: it suggests the directory was uploaded to an OPA server, when really it's just read and used.

While we're at it, let's drop the `/Users/...` prefix, maybe?

Reply:
Fixed.